kubernetes/token.go

```diff
@@ -10,29 +10,31 @@ import (
 
 var DefaultServiceAccountPath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
 
-var KialiToken string
+var KialiTokenForHomeCluster string
 var tokenRead time.Time
 
-func GetKialiToken() (string, error) {
+// GetKialiTokenForHomeCluster returns the Kiali SA token to be used to communicate with the local data plane k8s api endpoint.
+func GetKialiTokenForHomeCluster() (string, error) {
 	// TODO:refresh the token when it changes rather than after it expires
-	if KialiToken == "" || shouldRefreshToken() {
-		if remoteSecret, err := GetRemoteSecret(RemoteSecretData); err == nil {
-			KialiToken = remoteSecret.Users[0].User.Token
+	if KialiTokenForHomeCluster == "" || shouldRefreshToken() {
+		if remoteSecret, err := GetRemoteSecret(RemoteSecretData); err == nil { // for experimental feature - for when data plane is in a remote cluster
+			KialiTokenForHomeCluster = remoteSecret.Users[0].User.Token
 		} else {
 			token, err := os.ReadFile(DefaultServiceAccountPath)
 			if err != nil {
 				return "", err
 			}
-			KialiToken = string(token)
+			KialiTokenForHomeCluster = string(token)
 		}
 		tokenRead = time.Now()
 	}
-	return KialiToken, nil
+	return KialiTokenForHomeCluster, nil
 }
 
-// Check if token expired based on the kubernetes configuration
+// shouldRefreshToken checks to see if the local Kiali token expired.
+// TODO should check all tokens for all clusters
 func shouldRefreshToken() bool {
-
+	// TODO: hardcoded to 60s, do we want this configurable? Or do we need to obtain this from k8s somehow?
```

Comment:
This is just to prevent doing many reads from the disk, so maybe the configuration is not needed. There was a bit of discussion here: https://github.com/kiali/kiali/pull/5386#discussion_r957967702